### PR TITLE
DEV - versions of JS files written to a JS file to be included by loa…

### PR DESCRIPTION
**Pull request by jbrw**
This is a different(/simpler?) approach that attempts to solve the issue that https://github.com/discourse/discourse/pull/10575 was trying to address.

### CODE REVIEW

**Review by pmusaraj — APPROVED** [review]
This looks great, I like that the rake task writes to a JS file directly instead of injecting it via Rails.

File: app/assets/javascripts/discourse/app/lib/load-script.js [hunk 1]
```diff
@@ -1,6 +1,7 @@
 import { default as getURL, getURLWithCDN } from "discourse-common/lib/get-url";
 import { run } from "@ember/runloop";
 import { ajax } from "discourse/lib/ajax";
+import { PUBLIC_JS_VERSIONS } from "discourse/lib/public_js_versions";
```
Comment (eviltrout) [inline app/assets/javascripts/discourse/app/lib/load-script.js:4]: In the js world we use `kebab-case` (dasherize) for module names.

File: app/assets/javascripts/discourse/app/lib/load-script.js [hunk 2]
```diff
@@ -102,3 +107,17 @@ export default function loadScript(url, opts) {
     }
   });
 }
+
+export function cacheBuster(url) {
+  if (PUBLIC_JS_VERSIONS) {
+    const pathParts = url.split("/");
+    if (pathParts[1] === "javascripts") {
+      const version = PUBLIC_JS_VERSIONS[pathParts[2]];
+      if (version !== undefined) {
```
Comment (eviltrout) [inline app/assets/javascripts/discourse/app/lib/load-script.js:116]:
```suggestion
      if (typeof version !== "undefined") {
```
This is preferred due to the unusual case that undefined can be overwritten 🤪

File: test/javascripts/lib/load-script-test.js [hunk 3]
```diff
@@ -19,3 +20,28 @@ QUnit.skip(
     );
   }
 );
+
+QUnit.test("works when a value is not present", async (assert) => {
```
Comment (eviltrout) [inline test/javascripts/lib/load-script-test.js:24]: These two tests have `async` but are synchronous. Should be dropped.

**Review by eviltrout — APPROVED** [review]
This looks very nice! I like it a lot. My feedback is all minor.

**Pull request merged**